pinot-connectors/pinot-spark-2-connector/pom.xml

```diff
@@ -119,6 +120,47 @@
       <build>
         <plugins>
           <!-- scala build -->
+          <plugin>
+            <groupId>org.xolstice.maven.plugins</groupId>
+            <artifactId>protobuf-maven-plugin</artifactId>
+            <version>0.6.1</version>
+            <configuration>
+              <protocArtifact>com.google.protobuf:protoc:3.19.2:exe:${os.detected.classifier}</protocArtifact>
+              <pluginId>grpc-java</pluginId>
+              <pluginArtifact>io.grpc:protoc-gen-grpc-java:1.44.1:exe:${os.detected.classifier}</pluginArtifact>
+            </configuration>
+            <executions>
+              <execution>
+                <goals>
+                  <goal>compile</goal>
+                  <goal>compile-custom</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+          <plugin>
+            <artifactId>maven-shade-plugin</artifactId>
```

Comment:
we already have this in the spark-common pom wondering if we still need this here.
if it only works on concrete spark version pom then do we still need the shade plugin in common?

Reply:
Oh, you're right, great catch! This is already covered in the common.

I came up with this PR when comparing Spark2 and Spark3 pom files and got the wrong idea that this should be added to both. As a matter of fact, I need to remove shading from the Spark3 pom file as well.

I'll update the PR and resubmit.

Reply:
OK, after some testing, I think the best approach is to shade at the final (concrete) package level for two reasons:

- Even if we shade at the common package level and add dependency with `<classifier>shaded</classifier>`, `mvn` will try to bring unshaded versions of those libraries. We need to exclude them specifically.
- Shading at the 'common' package level creates an unnecessary encapsulation at the internal boundary between two Pinot packages

I moved the protoc call to the common package and shading logic to concrete packages. Let me know what you think

Reply:
yeah i thought so too let's shade on the final stage, it gives flexibility if at any time we need to change a specific version's shading strategy.

seen this in hadoop2-3 migration a while back